lib/open_pgp/encode/packet/body_chunk_impl.ex

```diff
@@ -0,0 +1,53 @@
+defimpl OpenPGP.Encode, for: OpenPGP.Packet.BodyChunk do
+  alias OpenPGP.Packet.BodyChunk
+
+  def tag(_), do: raise(".tag/1 not supported by design for #{inspect(@for)}.")
+
+  @doc """
+  Encode body chunk. Always uses new packet format.
+  Return encoded body chunk with the length header prefix.
+
+  ### Example:
+
+  Encodes one-octet length New Format Packet Length Header (up to 191 octets)
+
+      iex> OpenPGP.Encode.encode(%OpenPGP.Packet.BodyChunk{data: "Hello world!"})
+      <<12::8, "Hello world!">>
+
+  Encodes two-octet length New Format Packet Length Header (192-8383 octets)
+
+      iex> rand_bytes = :crypto.strong_rand_bytes(255)
+      ...> OpenPGP.Encode.encode(%OpenPGP.Packet.BodyChunk{data: rand_bytes})
+      <<192::8, 63::8, rand_bytes::binary>>
+
+  Encodes five-octet length New Format Packet Length Header (8384-4_294_967_295 octets)
+
+      iex> rand_bytes = :crypto.strong_rand_bytes(8384)
+      ...> OpenPGP.Encode.encode(%OpenPGP.Packet.BodyChunk{data: rand_bytes})
+      <<255::8, 8384::32, rand_bytes::binary>>
+  """
+  def encode(%BodyChunk{data: "" <> _ = data}, _opts) do
+    blen = byte_size(data)
+
+    hlen =
+      cond do
+        blen in 0..191 ->
+          <<blen::8>>
+
+        blen in 192..8383 ->
+          <<b1::8, b2::8>> = <<blen - 192::16>>
+          <<b1 + 192::8, b2::8>>
+
+        blen in 8384..0xFFFFFFFF ->
```

Comment:
These magic numbers should probably be module attributes for clarity. Would be especially awesome to include a link to RFC 2440 as a reference.

Reply:
That's right, those numbers are referenced in the `@doc` above, and I can't think of any meaningful name that would be descriptive enough...

```
Encodes one-octet length New Format Packet Length Header (up to 191 octets)
Encodes two-octet length New Format Packet Length Header (192-8383 octets)
Encodes five-octet length New Format Packet Length Header (8384-4_294_967_295 octets)
```

Reply:
I'll give this a try:

```elixir
  @one_octet_length 0..191
  @two_octet_length 192..8383
  @five_octet_length 8384..0xFFFFFFFF
```